0xPolygon/polygon-edge
# Update predeployed staking SC

# Description

Fixes EDGE-6

This PR updates the staking contract bytecode and sets the default maximum number of validators to max unit32.

# Changes include


# Checklist


## Testing


### Manual tests

1. Start --pos chain with default values for min/max validators and default values for min/max should be 1/4294967295
2. When creating genesys:
     - min-validator-count >= 1
     - min-validator-count <= max-validator-count
     - max-validator-count <= max.Uint32 (4294967295)
3. Can not add new validator if in the network there are max-validator-count validators
     1. Start pos network with 4 nodes and max-validator-count 4
     2. Try to add a new validator using staking (node 5), it will fail 
     3. Unstake from one of the validators
     4. Stake from 5th node, it will be successful
 4. Switch POA -> POS, there should be default values for min/max validators

# Documentation update
https://github.com/0xPolygon/polygon-edge-docs/pull/66

# Additional comments
Staking-contract PR (https://github.com/0xPolygon/staking-contracts/pull/10)
